api.go

```diff
@@ -398,74 +412,54 @@ func (p JobQuery) SetProgress(jobID string, progress string) error {
 // ProgressChanNoWait returns a receive only channel, and progress send to this channel will be set
 // Note that the error of the set progress is ignored
 // send to done channel cleans it up
-// the difference between this and ProgressChan is this will not wait for SetProgress to finish
-// in effect, it only sets the latest progress
-func (p JobQuery) ProgressChanNoWait(jobID string) chan<- string {
-	// defer func() {
-	//     fmt.Println("check progress exited")
-	// }()
-	pch := make(chan string)    // receives all progress
-	lpch := make(chan string)   // receives only latest progress
+// the difference between this and ProgressChan is this will not wait for SetProgress to finish.
+func (p *JobQuery) ProgressChanNoWait(jobID string) chan<- string {
+	ch := make(chan string)
+	cch := make(chan string, 100)
 	done := make(chan struct{}) // signal no more progress is here
 
 	p.wg.Add(1)
-	// go routine waiting for the latest progress
 	go func() {
 		defer p.wg.Done()
 		for {
 			select {
-			case progress := <-lpch:
+			case progress := <-cch:
 				if progress == "" {
 					continue
 				}
-				// println("setting progress", jobID, progress)
 				err := p.SetProgress(jobID, progress)
 				if err != nil {
 					log.WARNING.Printf("failed to set progress for job id %s: %s, err: %v", jobID, progress, err)
 				}
 			case <-done:
 				return
+			default:
```

Comment:
this discards the latest progress, where the original implementation discards the old ones and it also does deduplication